http3/roundtrip.go

```diff
@@ -41,11 +41,23 @@ type RoundTripper struct {
 	// If nil, reasonable default values will be used.
 	QuicConfig *quic.Config
 
+	// Settings specifies the HTTP/3 settings transmitted to the server.
+	// If nil, reasonable default values will be used.
+	// See https://quicwg.org/base-drafts/draft-ietf-quic-http.html#name-http-2-settings-parameters.
+	// If non-nil, overrides EnableDatagrams and EnableWebTransport.
```

Comment:
Why do we need this? I don't like the fact that this overrides other options.
Do we need to export `Settings` at all? Ideally, the user of the library wouldn't have to deal with setting code points at all.

Reply:
My preference is to remove `EnableDatagrams` and `EnableWebTransport` in favor of letting the caller specify `Settings`.

The `Settings` type has helpers for enabling datagrams or WebTransport, or other common settings.